lms/djangoapps/monitoring/scripts/generate_code_owner_mappings.py

```diff
@@ -3,61 +3,149 @@
 
-Sample CSV input::
+Or for more details::
 
-    Path,owner.squad
-    ./common/djangoapps/xblock_django,team-red
-    ./openedx/core/djangoapps/xblock,team-red
-    ./lms/djangoapps/badges,team-blue
+    python lms/djangoapps/monitoring/scripts/generate_code_owner_mappings.py --help
 
-Sample output::
-
-    # Copy results into appropriate config yml file.
-    CODE_OWNER_MAPPINGS:
-      team-blue:
-      - badges
-      team-red:
-      - xblock_django
-      - openedx.core.djangoapps.xblock
 
 """
-import os
-import re
 import csv
 import click
+import os
+import re
+
+# Maps edx Django apps to their containing repo
+EDX_REPO_APPS = {
+    'bulk_grades': 'edx-bulk-grades',
+    'coaching': 'platform-plugin-coaching',
+    'completion': 'completion',
+    'config_models': 'django-config-models',
+    'consent': 'edx-enterprise',
+    'csrf': 'edx-drf-extensions',
+    'edx_proctoring': 'edx-proctoring',
+    'edxval': 'edx-val',
+    'enterprise': 'edx-enterprise',
+    'enterprise_learner_portal': 'edx-enterprise',
+    'help_tokens': 'help-tokens',
+    'integrated_channels': 'edx-enterprise',
+    'organizations': 'edx-organizations',
+    'search': 'edx-search',
+    'wiki': 'django-wiki',
+}
+
+# Maps third-party Django apps to their containing repo
+THIRD_PARTY_APPS = {
+    'django': 'django',
+    'django_object_actions': 'django-object-actions',
+    'drf_yasg': 'drf-yasg',
+    'lx_pathway_plugin': 'lx-pathway-plugin',
+    'simple_history': 'django-simple-history',
+    'social_django': 'social-app-django',
+}
 
 
 @click.command()
+@click.option(
+    '--repo-csv',
+    help="File name of .csv file with repo ownership details.",
+    required=True
+)
 @click.option(
     '--app-csv',
     help="File name of .csv file with edx-platform app ownership details.",
     required=True
 )
-def main(app_csv):
+@click.option(
+    '--dep-csv',
+    help="File name of .csv file with edx-platform 3rd-party dependency ownership details.",
+    required=True
+)
+def main(repo_csv, app_csv, dep_csv):
     """
     Reads CSV of ownership data and outputs config.yml setting to system.out.
 
-    Expected CSV format:
+    Expected Repo CSV format:
 
-        Path,owner.squad\n
-        ./lms/templates/oauth2_provider,team-red\n
-        ./openedx/core/djangoapps/user_authn,team-blue\n
+        \b
+        repo name,owner.squad
+        edx-platform,team-red
+        edx-proctoring,team-blue
+        ...
+
+    Expected App CSV format:
+
+        \b
+        Path,owner.squad
+        ./lms/templates/oauth2_provider,team-red
+        ./openedx/core/djangoapps/user_authn,team-blue
+        ...
+
+    Expected 3rd-party Dependency CSV format:
+
+        \b
+        repo name,repo url,owner.squad
+        django,team-red
+        social-app-django,team-blue
         ...
 
     Final output only includes paths which might contain views.
 
     """
-    csv_data = None
-    with open(app_csv, 'r') as file:
+    owner_to_paths_map = {}
```

Comment:
So this is a map of owning squad names to lists of dotted module paths?

Reply:
Added comment for clarity.